protocols/plaintext/src/handshake.rs

```diff
@@ -109,45 +104,33 @@ impl HandshakeContext<Local> {
     }
 }
 
-pub fn handshake<S>(socket: S, config: PlainText2Config)
-    -> impl Future<Item = (Framed<S, BytesMut>, Remote), Error = PlainTextError>
+pub async fn handshake<S>(socket: S, config: PlainText2Config)
+    -> Result<(Framed<S, unsigned_varint::codec::UviBytes<Vec<u8>>>, Remote), PlainTextError>
```

Comment:
I would import `unsigned_varint::codec::UviBytes` and shorted this line. Again, I would keep using `BytesMut`.